src/icon.c

```diff
@@ -161,18 +161,18 @@ cairo_surface_t *icon_get_for_notification(const struct notification *n)
         int h = gdk_pixbuf_get_height(pixbuf);
         int larger = w > h ? w : h;
         if (settings.max_icon_size && larger > settings.max_icon_size) {
-                GdkPixbuf *scaled;
-                if (w >= h) {
-                        scaled = gdk_pixbuf_scale_simple(pixbuf,
-                                        settings.max_icon_size,
-                                        (settings.max_icon_size * h) / w,
-                                        GDK_INTERP_BILINEAR);
-                } else {
-                        scaled = gdk_pixbuf_scale_simple(pixbuf,
-                                        (settings.max_icon_size * w) / h,
-                                        settings.max_icon_size,
-                                        GDK_INTERP_BILINEAR);
-                }
+                int scaled_w = settings.max_icon_size;
+                int scaled_h = settings.max_icon_size;
+                if (w>=h)
+                        scaled_w = (settings.max_icon_size * h) / w;
```

Comment:
Opps! Sorry I missed this earlier: You've reversed this, if `width >= height` then the width should be scaled down to `max_icon_size`, and we have to calculate the appropriate height.

And do the reverse if `height > width`.